File: 1.Introduction/program.js
const number = 5;

let [func1Output, func2Output, func3Output] = [0, 0, 0]

const findSum = (number) => {
  func1Output = func1(number);
  func2Output = func2(number);
  func3Output = func3(number);
}

function func1(number) {
  return (number * (number + 1)) / 2
}

function func2(number) {
  let sum = 0
  for (let i = 1; i <= number; i++) {
    sum += i;
  }
  return sum;
}

function func3(number) {
  let sum = 0;
  for (let i = 1; i <= number; i++) {
    for (let j = 1; j <= i; j++) {
      sum+=j;
    }
  }
  return sum;
}

function func3(number) {
  let sum = 0;
  for (let i = 1; i <= number; i++) {
    for (let j = 1; j <= i; j++) {
      sum ++;
    }
  }
  return sum;
}

findSum(number)

console.log(`func1Output: ${func1Output}, func2Output: ${func2Output}, func3Output: ${func3Output}`)

File: 1.Introduction/program.py
number = 5

func1Output, func2Output, func3Output = 0, 0, 0

def findSum(number):
    global func1Output, func2Output, func3Output
    func1Output = func1(number)
    func2Output = func2(number)
    func3Output = func3(number)

def func1(number):
    # Sum of the first 'number' natural numbers using the formula
    return (number * (number + 1)) / 2

def func2(number):
    # Sum from 1 to 'number' (inclusive)
    sum = 0
    for i in range(1, number + 1):  # start at 1, go up to 'number'
        sum += i
    return sum

def func3(number):
    # Sum of all numbers below each number up to 'number'
    total_sum = 0
    for i in range(1, number + 1):
        for j in range(i):  # sum from 0 to i-1
            total_sum+=1
    return total_sum

findSum(number)

print(f"func1Output: {func1Output}, func2Output: {func2Output}, func3Output: {func3Output}")
